osquery/tables/system/tests/darwin/asl_tests.cpp

```diff
@@ -174,33 +175,39 @@ TEST_F(AslTests, test_convert_like_regex) {
 
 TEST_F(AslTests, test_actual_query) {
   auto version = SQL::selectAllFrom("os_version");
-  auto minor_version_exp = tryTo<unsigned long int>(version[0]["minor"], 10);
-  ASSERT_TRUE(minor_version_exp.isValue());
-  if (minor_version_exp.get() >= 12) {
-    // macOS Sierra and above do not support ASL.
+  auto minor_version = tryTo<unsigned long int>(version[0]["minor"], 10);
+  auto major_version = tryTo<unsigned long int>(version[0]["major"], 10);
+  ASSERT_TRUE(minor_version.isValue());
+  ASSERT_TRUE(major_version.isValue());
+
+  // In macOS before 10.12, it was possible to write to the ASL using
+  // '/usr/bin/logger', but since 10.12, invoking 'logger' sends the log entry
+  // to the Unified Logging system, and ASL is deprecated. Because osquery
+  // actually reads the syslog downstream of ASL, not ASL logs directly,
+  // we can use syslog() to write a log entry and then read it back via
+  // the `asl` virtual table.
```

Comment:
Outside the context of this PR, I'm not sure this comment clarifies anything -- post merge there is no `logger`, so we don't need to justify it's absence.

Reply:
great point, I've changed it to make it relevant